lib/tools/lockmgmt.js

```diff
@@ -104,4 +107,41 @@ lockManagementMethods.isLockEnabled = async function isLockEnabled () {
   }
 };
 
+/**
+ * Sets the device lock.
+ *
+ * @param {!string} credentialType One of: password, pin, pattern.
+ * @param {!string} credential A non-empty credential value to be set.
+ * Make sure your new credential matches to the actual system security requirements,
+ * e.g. a minimum password length. A pattern is specified by a non-separated list
+ * of numbers that index the cell on the pattern in a 1-based manner in left
+ * to right and top to bottom order, i.e. the top-left cell is indexed with 1,
+ * whereas the bottom-right cell is indexed with 9. Example: 1234.
+ * @param {?string} oldCredential [null] An old credential string.
+ * It is only required to be set in case you need to change the current
+ * credential rather than to set a new one. Setting it to a wrong value will
+ * make this method to fail and throw an exception.
+ * @throws {Error} If there was a failure while verifying input arguments or setting
+ * the credential
+ */
+lockManagementMethods.setLockCredential = async function setLockCredential (
+  credentialType, credential, oldCredential = null) {
+  if (!SUPPORTED_LOCK_CREDENTIAL_TYPES.includes(credentialType)) {
+    throw new Error(`Device lock credential type '${credentialType}' is unknown. ` +
+      `Only the following credential types are supported: ${SUPPORTED_LOCK_CREDENTIAL_TYPES}`);
+  }
+  if (_.isEmpty(credential) && !_.isInteger(credential)) {
```

Comment:
should `password` case also be an integer?

Reply:
no, it's just in case the one passes an integer while string is expected. isEmpty returns true in such case, however it is still possible to apply such value

Reply:
ah... My thought was wrong... (i noticed...)
and didn't know `isEmpty(123)` etc returned true...